Limit milliseconds to three digits

stoiveyp opened the pull request:
Fixes #724 and adds test to ensure that's all that changes
No standard format limits to three digits for milliseconds, but the format is the same as `"o"` used before in all other regards

twsouthwick commented:
@tomjebo can you review this as it's more up your alley. Is the form consistent with the spec? Let's add the section from the spec for reference

twsouthwick commented:
@stoiveyp can you update the changelog to include this change? You may need to merge the latest changes from master if you don't have a 2.11.1 spot for the changes.

twsouthwick commented on src/DocumentFormat.OpenXml/Spreadsheet/CellValue.cs:
```diff
@@ -17,7 +17,7 @@ public partial class CellValue
         /// <param name="dateTime">DateTime for cell</param>
         public CellValue(DateTime dateTime)
-            : this(dateTime.ToString("o", CultureInfo.InvariantCulture))
+            : this(dateTime.ToString("yyyy'-'MM'-'dd'T'HH':'mm':'ss'.'fff", CultureInfo.InvariantCulture))
```
I'd prefer if this was a const, ie

```cs
private const string DateTimeFormatString = "....";
private const string DateTimeOffsetFormatString = DateTimeFormatString + "zzz";
```

It would make it much easier to see that they're pretty much the same.

twsouthwick commented:
Also, would it make sense to provide a parse function? We want to get a revision version out which we don't expose new APIs, but we could expose it on the next minor version

twsouthwick commented:
Could we use `XmlConvert.ToString` instead? Wouldn't that be the same, but not require a custom format string

stoiveyp commented:
@twsouthwick sorry for the delay - day job has worn me out this week! 
Did look at XmlConvert, but it also goes down to too many significant digits in the seconds fraction, so we end up having to customise there too.
I'll make the other changes now though 👍

stoiveyp commented:
@twsouthwick moved the string conversion to static methods in CellValue and moved the formats to constants
Changelog is also updated


Can't find anything in the OpenXml spec specifically about date time cell values.
So I looked in the W3 Xml Schema documents
https://www.w3.org/TR/xmlschema-2/
```
All ·minimally conforming· processors ·must· support year values with a minimum of 4 digits (i.e., YYYY) and a minimum fractional second precision of milliseconds or three decimal digits (i.e. s.sss). However, ·minimally conforming· processors ·may· set an application-defined limit on the maximum number of digits they are prepared to support in these two cases, in which case that application-defined maximum number ·must· be clearly documented.
```

So without comment within the OpenXml spec or Excel documentation - a processor based on the Xml Schema spec would only have to support the three digits we're now outputting.

twsouthwick commented:
LGTM! I've opened #752 to add a parsing method so people don't need to know the value to get a strongly typed value. That will have to go into a minor version (which may end up being the next version... there's a few APIs we'll want to expose soon).
